docs/lesson-plans/week-4.md

```diff
@@ -0,0 +1,623 @@
+port range, and ports :3000, :8000, and :8081 (what we'll be using) are quite common. Also, when testing locally,
+HTTPS is usually not used since a certificate has to be created, so just focus on using HTTP for now. We'll talk about
+HTTP more later.
+
+To access a port at a given address, you write the protocol (optional, http is assumed by default), followed by the 
+address, followed by the colon-prefixed port. So for example, accessing http on port :4000 locally would look something
+like `http://localhost:4000` or `http://127.0.0.1:4000`.
+
+#### Ngrok
+
+When you're hosting on localhost, it's often not easy for someone on a different network or even the same network to
+see the progress you've made and access your app. This is because of a bunch of firewall rules in place, and to fix that
+you will end up having to look into to port-forward your application or messing with those firewall rules. An easy way
+to get around that is to look into [Ngrok](https://ngrok.com/), which ends up forwarding traffic through one of their 
+servers for your local app and avoids having to deal with those rules. So if you want to show your friends and family,
+download it and check out their [docs](https://ngrok.com/docs). It's really easy and extremely useful!
+
+## HTTP Requests and REST
+
+### What is HTTP
+
+Hypertext Transfer Protocol (HTTP) is a protocol for requests and responses in a server-client architecture model. In
+the web browser example from earlier, the client's request to the server is an HTTP request, and the server provides an
+HTTP response in the form of actions and resources like HTML files. 
+
+HTTP provides a few useful items of information we'll be using such as headers in both requests and responses, which 
+provide metadata about the request or response, bodies, which contain the actual information to be transferred if any, 
+and status codes, which let you know if any actions were performed, errors occurred, and what the response means.
+
+#### Status Codes in HTTP
+
+Status codes are special numerical codes returned in an HTTP response which indicate the status of the request. There 
+are a couple groups of status codes, which define a high level meaning as to what happened during the server.
+
+| Code Range | Description |
+| :--- | :--- |
+| 100 - 199 | Informational Responses |
+| 200 - 299 | Successful Responses |
+| 300 - 399 | Redirect Responses |
+| 400 - 499 | Client Errors |
+| 500 - 599 | Server Errors |
+
+For more info on the responses outside of what's below, check out [MDN's HTTP Status Code 
+Guide](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status) or the [API REST Tutorial 
+Cheat Sheet](https://www.restapitutorial.com/httpstatuscodes.html).
+
+You can also see a lot of these if you open up your browser's developer console and look at the network tab. This 
+network tab can give you information about exactly what's going on when you try to load a webpage. Try it out!
```

Comment:
Add a short note on how they can do this! I know on Mac you just right click and click Inspect Element but I'm not sure how to do so on Windows. I'm assuming its the same?

Reply:
I second this

Reply:
I'm just going to do the `Inspect/Inspect Element` one because it's a lot easier to say